src/components/IconState.vue

```diff
@@ -0,0 +1,79 @@
+<template>
+  <span class="icon-state">
+    <v-icon color="primary" dark @click.stop="dialog = true">{{ localIcon }}</v-icon>
+    <v-dialog v-model="dialog" max-width="400">
+      <v-card>
+        <v-card-title class="headline">Please choose your Icon?</v-card-title>
+        <v-card-text>
+          <v-row>
+            Please enter the name of icon you wish to use for this state. You can get the icon names from
+                  <v-chip
+                    class="mr-2"
+                    @click="openMaterial"
+                  >
+                    <!-- could not find a apporprite icon.-->
+                    Material
+                  </v-chip>
+                  <v-chip
+                    class="mr-2"
+                    rounded
+                    @click="openMaterialDesign">
+                      <!-- could not find a apporprite icon.-->
+                      Material desgin
+                  </v-chip>
+          </v-row>
+          <v-row>
+            <div id="local-icon">
+              <v-icon>{{localIcon}}</v-icon>
+            </div>
+            <v-text-field v-model="localIcon"></v-text-field>
+          </v-row>
+        </v-card-text>
+        <v-card-actions>
+          <v-spacer></v-spacer>
+          <v-btn color="green darken-1" text @click="close">Nah...</v-btn>
```

Comment:
Just a general comment here - the theme is going to change soon, with #9 . These colors are going to be removed, and should be replaced with `primary`, `secondary`, 'accent', etc. This should probably be `accent`. See the theme guidelines for Vuetify - https://dev.vuetifyjs.com/en/customization/theme/